src/components/Sidebar/SearchBox.test.js

```diff
@@ -21,19 +21,45 @@ describe('SearchBox', () => {
   });
 
   it('renders text typed into the search box', () => {
+    // const handleSearchInputMock = jest.fn();
+    const classes = mockStyles(styles);
+    const wrapper = shallow(<UnstyledSearchBox
+      classes={classes}
+      handleSearchInput={() => {}}
+      clearSearchResults={() => {}}
+    />);
+
+    const testSearchInput = 'EECS';
+    const eventObject = makeEventObject(testSearchInput);
+    wrapper.find('TextField').simulate('change', eventObject);
+
+
+    // expect(handleSearchInputMock).toHaveBeenCalledWith(testSearchInput);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('doesn\'t call api unless it has been 300ms since last keystroke', (completed) => {
     const handleSearchInputMock = jest.fn();
+
     const classes = mockStyles(styles);
     const wrapper = shallow(<UnstyledSearchBox
       classes={classes}
       handleSearchInput={handleSearchInputMock}
       clearSearchResults={() => {}}
     />);
 
+    expect(handleSearchInputMock).not.toBeCalled();
+
     const testSearchInput = 'EECS';
     const eventObject = makeEventObject(testSearchInput);
     wrapper.find('TextField').simulate('change', eventObject);
-    expect(handleSearchInputMock).toHaveBeenCalledWith(testSearchInput);
-    expect(wrapper).toMatchSnapshot();
+
+    expect(handleSearchInputMock).not.toBeCalled();
+
+    setTimeout(((callback) => {
```

Comment:
That `callback` needs to be passed into the interior of the `setTimeout` function because this is a Jest async test, so it will end as soon as the setTimeout function returns (immediately) and ignore the `expect` call within the setTimeout if it isn't setup this way. With this extra function passed in, Jest knows not to officially "end" the test until that function has been called (in this case, within the body of `setTimeout`. I'll add a comment noting this.

Reply:
Let's also change the name of `callback()` to something more specific, like `jestCompleted`.